includes/utilities.inc

```diff
@@ -20,12 +20,15 @@ function islandora_usage_stats_log_object_view(AbstractObject $object) {
       $pid_id = islandora_usage_stats_get_pid_id($object->id);
       $object_access_id = islandora_usage_stats_object_view_to_db($pid_id);
 
+      module_invoke_all('islandora_usage_stats_access_recorded', 'object', $object_access_id, $pid_id, $object->id);
+
       // Log content models.
       foreach ($object->models as $model) {
         if ($model != 'fedora-system:FedoraObject-3.0') {
           try {
             $model_id = islandora_usage_stats_get_pid_id($model);
             islandora_usage_stats_content_model_view_to_db($object_access_id, $model_id);
+            module_invoke_all('islandora_usage_stats_access_recorded', 'cmodel', $model_id);
```

Comment:
This here is confusing -could lead to issues-. The new hook is no CMODEL specific hook, so actually the same hooks will be invoked everytime for every CMODEL. And all will depend on how smart the logic inside those is to avoid making mistakes/recording too many times, or losing computing cycles (feel old when i say that.. i know..). See my previous comment.

Reply:
Efficiency is important in this hook, so if providing a way to implement cmodel-specific implementations is more efficient, I can't argue with that. So you're suggesting something like defining `hook_cmodel_pid_islandora_usage_stats_access_recorded()`?

Reply:
@DiegoPino thinking about this a bit more, I want to clarify that this hook doesn't inspect the cmodel of the object (which is what other `hook_cmodel_pid_i` hooks do), it fires when the base module records an access on the cmodel. Don't know if that's relevant to this discussion, just thought I'd mention it.

Reply:
I would agree with @DiegoPino that if there IS a way to broadcast a generic hook (for all cmodels) AND a hook that is specific for cmodels, that would be preferred in case someone wanted to implement a hook, but only when citations are viewed for example.

Reply:
OK, but if the cmodel is just a string value that indicates what is being counted (that is all it is), does supplying cmdodel-specific hook definitions really matter? I just want to make sure that I'm not misunderstanding something here.

Reply:
Lets say that we want to record the browser used for everyone viewing books. In the current situation, we would need to implement a hook and check to see what cmodel it is, and if its a book THEN do some extra behavior to log the user's browser somewhere. This wouldn't take a lot of time, but it would fire for EVERY OBJECT. 

Alternatively, if there was an `access_recorded` hook that just fired when books were being recorded, you would shave that time off of having the hook fire for non-book objects.

Now, as I write this, I'm also wondering if this use case couldn't also be supported by hooking into `islandora_object_view` or something similiar, which I believe usage stats is hooking into. Maybe this IS overkill. Shrug.

Reply:
I don't think that would be overkill.